File: woodroid/src/main/java/me/gilo/woodroid/data/TaxClassAPI.java
package me.gilo.woodroid.data;


import me.gilo.woodroid.models.Coupon;
import me.gilo.woodroid.models.TaxClass;
import retrofit2.Call;
import retrofit2.http.*;

import java.util.List;

public interface TaxClassAPI {

    @Headers("Content-Type: application/json")
    @POST("taxes/classes")
    Call<TaxClass> create(@Body TaxClass body);

    @GET("taxes/classes")
    Call<List<TaxClass>> list();

    @DELETE("taxes/classes/{id}")
    Call<TaxClass> delete(@Path("id") int id);

    @DELETE("taxes/classes/{id}")
    Call<TaxClass> delete(@Path("id") int id, @Query("force") boolean force);

}

File: woodroid/src/main/java/me/gilo/woodroid/data/api/TaxClassAPI.kt
package me.gilo.woodroid.data.api


import me.gilo.woodroid.models.TaxClass
import retrofit2.Call
import retrofit2.http.*

import java.util.ArrayList

interface TaxClassAPI {

    @Headers("Content-Type: application/json")
    @POST("taxes/classes")
    fun create(@Body body: TaxClass): Call<TaxClass>

    @GET("taxes/classes")
    fun list(): Call<List<TaxClass>>

    @DELETE("taxes/classes/{id}")
    fun delete(@Path("id") id: Int): Call<TaxClass>

    @DELETE("taxes/classes/{id}")
    fun delete(@Path("id") id: Int, @Query("force") force: Boolean): Call<TaxClass>

    @GET("coupons")
    fun filter(@QueryMap filter: Map<String, String>): Call<ArrayList<TaxClass>>

}